Fixed monitoring for cron job

xFaNaTiix opened the pull request:
This should fix "monitor" when it's run by a cron job or something like this. (e.g. #318)
Please take a look at it. It works for me. =)

dgibbs64 commented:
Interesting didn't realise there was a problem. I run monitor and haven't had an issue before. What distro you running? Also any ideas why this is an issue for you? Im happy to merge once i have tested but im curious on why this is an issue.

xFaNaTiix commented:
Distro Details:
Distro:    Ubuntu 14.04.2 LTS
Arch:      x86_64
Kernel:    3.13.0-35-generic
Hostname:  puck372
tmux:      tmux 1.8
GLIBC:     2.19


*/5 * * * * /home/marvin/gameservers/csgoserver monitor > /dev/null 2>&1

If I run "./csgoserver monitor" it works fine but If this command is run by a cron job:
Mai 23 17:26:02 csgo-server: Monitor: Disabled: No lock file found

Apparently there are path issues. So I changed the path to the full path. https://github.com/xFaNaTiix/linuxgsm/blob/6bb595cc33d053e5f25966a91d57c68b10d384fb/functions/fn_monitor#L64

Then the first issue was gone but then:
Mai 23 17:27:03 csgo-server: Monitor: Checking session: CHECKING
Mai 23 17:27:04 csgo-server: Monitor: Checking session: OK

..it didn't start gsquery.py because it didn't find it.

So I changed all paths to the full paths and it worked (fn_monitor_query) :)

Scarsz commented:
Commits look good and even if it's not an issue for some people it's in good practice to have full pathnames to files.
